### ISSUE TITLE
Are there significant differences in precision and performance when deploying smoothquant using the ONNX/TRT path and the TensorRT-LLM path?

### ISSUE THREAD

**Issue by tp-nan** [NVIDIA/TensorRT-Model-Optimizer#11]
Hello, I am considering deploying a quantized model similar to llama2 using smoothquant. I am exploring two options: using ONNX and TRT-LLM. 

1. Would there be any difference in precision when accelerating with TensorRT after converting to ONNX, compared to accelerating with TRT-LLM? 

2. For some specific reasons, we are not currently considering aspects such as kvcache and scheduling. Then would there be any performance difference between using TensorRT and TensorRT-LLM? Thanks.

**Comment by cjluo-nv** [comment]
Sure. Please see the comments inline:

Would there be any difference in precision when accelerating with TensorRT after converting to ONNX, compared to accelerating with TRT-LLM?
In an idea situation, there should not be. However, the detailed kernels that the two paths use are different. And TRT-LLM also has some customized kernels for LLM specific support. So it's possible that the two paths has a slight precision difference.

For some specific reasons, we are not currently considering aspects such as kvcache and scheduling. Then would there be any performance difference between using TensorRT and TensorRT-LLM?
TLDR: If you are working on LLM, we always recommend to use TRT-LLM for deployment as it applies LLM specific optimizations like KV caching, inflight-batching and many more for LLM specific usecases.

**Comment by nan2088** [comment]
thx for your kindly explaination.
